csharp/HOLMS.Platform/HOLMS.Platform/Application.Client/ApplicationClient.cs

```diff
@@ -278,7 +305,13 @@ public class ApplicationClient : IDisposable, IApplicationClient {
                 GrantType = "refresh_token",
                 RefreshToken = SC.RefreshToken,
             };
-            var sessionResponse = await SessionSvc.RefreshAccessTokenAsync(refreshRequest, new CallOptions(null, DateTime.UtcNow.AddSeconds(30)));
+
+            var refreshChannel = new Channel($"{_sp.AppSvcHostname}:{_sp.AppSvcPort}",
+                AccessToken.NullToken.ToChannelCredentials());
+            var ss = new SessionSvc.SessionSvcClient(refreshChannel);
```

Comment:
is there any value in trying to de-duplicate this code? I don't think it's reasonable due to performing the shutdown on the channel and needing the service call.